src/txn/txn.c

```diff
@@ -1453,7 +1454,8 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
              */
             if (!commit && resolve_case == RESOLVE_PREPARE_ON_DISK &&
               head_upd->type == WT_UPDATE_STANDARD && F_ISSET(head_upd, WT_UPDATE_RESTORED_FROM_HS))
-                WT_ASSERT(session, head_upd->next == NULL);
+                WT_ASSERT_ALWAYS(session, head_upd->next == NULL,
+                  "Failed to correctly restore update from history store");
```

Comment:
I'm not sure what this assert is protecting against.. It's either that we've restored multiple updates from the history store or we've placed the history update at the wrong location in the chain

Reply:
We're rolling back a prepared transaction as indicated by the `!commit` check, I'd say that this is ensuring that we correctly placed the update prior to the prepared transaction on the update chain. As it was restored from the history store it wouldn't make sense to have something "older" than it on the chain. 

Perhaps we can word it as follows: "Rolling back a prepare transaction resulted in an invalid update chain."